From 487c53d3b909cbee287e153bc19885637235f495 Mon Sep 17 00:00:00 2001
From: Zhou weixin <zwx@rock-chips.com>
Date: Wed, 17 Aug 2016 10:18:30 +0800
Subject: [PATCH] pwm: rockchip: Make pwm polarity to be configured correctly

If pwm polarity was configured with different values at uboot,
the enable_conf would not be configured correctly.

Change-Id: I55b9ccc262382951a8a82810f1be74ce9460f266
Signed-off-by: Zhou weixin <zwx@rock-chips.com>
---
 drivers/pwm/pwm-rockchip.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pwm/pwm-rockchip.c b/drivers/pwm/pwm-rockchip.c
index f11e1fc9f97b..beab624fb9d9 100644
--- a/drivers/pwm/pwm-rockchip.c
+++ b/drivers/pwm/pwm-rockchip.c
@@ -89,6 +89,7 @@ static void rockchip_pwm_set_enable_v2(struct pwm_chip *chip,
 		enable_conf |= PWM_DUTY_POSITIVE | PWM_INACTIVE_NEGATIVE;
 
 	val = readl_relaxed(pc->base + pc->data->regs.ctrl);
+	val &= ~(GENMASK(5, 0) | BIT(8));
 
 	if (enable)
 		val |= enable_conf;
-- 
2.35.3

